Suggest missing "symfony/yaml" dependency

JanWennrich opened the pull request:
To use the YAML source type, the `symfony/yaml` package is required.

JanWennrich commented:
As an alternative to requiring the package, it could be added as a suggestion to the `composer.json`.  
If that is done, the `README` should state in the `source types` section, that the package is required.

Crell commented:
It is deliberately optional, so that if you don't want to use Yaml you don't need to install a useless package.

Adding it to the suggestions block and README is fine, though.  PR welcome for that.  (Though, I though the README already said that somewhere, doesn't it?)

JanWennrich commented:
Hey @Crell,

I just reverted my previous commit.  
Then I turned the requirement into a suggestion and added a note to the `README`.

Furthermore I'd like to know your opinion on the following things:
- Should I squash/drop my previous requirement commit and the revert commit (c67f21f3d0581df86a248dc6cfb0fb8be936eb16, 45c8621b82d4ea98e7ebe547dbf6ded5f1f9c10d)?
- Should the examples in the `README` use a source type that works without the `symfony/yaml` requirement? This would make a "quick start" or testing the library easier.
- Is it necessary to add a conflict to the `composer.json` for non-supported versions of the `symfony/yaml` library?

Crell commented:
1. Yes, please rebase/squash out the old commits.  Or I can just squash-merge this whenever it gets merged.
2. Hm, yeah, I guess it makes sense to switch to JSON for the example block.  Though in practice I figure most people will use YAML, even if YAML is horrible. :smile: 
3. I don't think so?  Let's skip a conflict right now.  Technically I think it would work with a 5.x version of symfony/yaml, since the base API hasn't changed in a decade.  Just because a recent version is all I can be bothered to test with doesn't mean older versions can't work. :smile: 
4. The PHPStan failure looks unrelated.  Probably a bugfix in PHPStan that we're getting hit by.  (This happens often.)  If you're able to fix it as part of this PR, go for it.  It should only need adding a docblock type.  If you're not sure, let me know and I'll see if I can find time to fix it in `master`.

Crell commented:
Also, it's great to see someone take interest in this library. :stuck_out_tongue: I didn't think anyone else was using it.

JanWennrich commented:
Hi @Crell,

1. I dropped the old commits
2. I will open a different pull request for that
3. I agree, let's ignore the conflicts
4. The PHPStan error should be fixed (on my machine at least). I specified the types based on the Serde attributes that should only allow those three classes as values and only strings as keys.

Crell commented on README.md:
```diff
@@ -59,7 +59,12 @@ You can and should define as many different config objects as you'd like.  They
 
-There are several file formats supported out of the box, including `JsonFileSource`, `YamlFileSource`, `PhpFileSource`, and even `IniFileSource` (because why not?).  Writing other sources is simple, as the `ConfigSource`interface has only a single method.  You can even stack multiple file types with the same directory to read from into a single list, if you want to support multiple file types.
+There are several file formats supported out of the box, including `JsonFileSource`, `YamlFileSource`, `PhpFileSource`, and even `IniFileSource` (because why not?).  
+Note: In order to use the `YamlFileSource` you need to have the [`symfony/yaml`](https://packagist.org/packages/symfony/yaml) installed. 
```
```suggestion
Note: In order to use the `YamlFileSource` you need to have the [`symfony/yaml`](https://packagist.org/packages/symfony/yaml) package installed. 
```
